src/main/java/uk/ac/ebi/ega/egafuse/service/EgaDirectory.java

```diff
@@ -83,7 +85,12 @@ public synchronized void read(Pointer buf, FuseFillDir filler) {
             if (getName().equalsIgnoreCase("datasets")) {
                 egaDatasetService.getDatasets().forEach(this::add);
             } else {
-                egaFileService.getFiles(this).forEach(this::add);
+                List<EgaFile> egaFiles = egaFileService.getFiles(this);
+
+                if (isTreeStructureEnable)
+                    egaDatasetService.buildSubDirectoryFromFilePath(egaFiles, this);
+                else
+                    egaFiles.stream().forEach(this::add);
```

Comment:
I think overriding the add method or creating a new one addAll would be more convenient than calling the single add many times

Reply:
We are also setting the parent value for each file in the add method. So can't use addall, we would still need to loop around each file and add it.